### PR TITLE
Add missing expansion patterns for `math.powf`

### LINKED ISSUE #13547: [CPU] Vector powf is not vectorized
It looks like vector pow operations are not lowered to actual vector operations (not sure if only in some cases). In particular, we generate the following MLIR when targeting AVX512:

```
      %10 = math.powf %9, %cst_18 : vector<8x32xf32>
```

which is eventually lowered to a `powf` function call (perf):

```
  37.17%  iree-benchmark-  iree_dylib_p0fvBe_mem_.so  [.] powf                                                                                                          
```

implemented with scalar instructions:

```
  0.02 │       vmulsd    tfcompile_1038_linked_llvm_cpu+0xa20, %xmm0, %xmm4
  0.28 │       vaddsd    tfcompile_1038_linked_llvm_cpu+0xa98, %xmm4, %xmm4
  0.79 │       vmulsd    tfcompile_1038_linked_llvm_cpu+0xa90, %xmm0, %xmm5
  1.24 │       vaddsd    tfcompile_1038_linked_llvm_cpu+0xa80, %xmm5, %xmm5
  0.02 │       vmulsd    %xmm3, %xmm3, %xmm6
  0.17 │       vmulsd    %xmm6, %xmm4, %xmm4
  0.84 │       vmulsd    tfcompile_1038_linked_llvm_cpu+0xa78, %xmm0, %xmm0
  1.13 │       vaddsd    %xmm0, %xmm2, %xmm0
  0.02 │       vmulsd    %xmm5, %xmm3, %xmm2
  0.18 │       vaddsd    %xmm2, %xmm0, %xmm0
  0.90 │       vaddsd    %xmm0, %xmm4, %xmm0
  1.83 │       vcvtss2sd %xmm1, %xmm1, %xmm1
  0.15 │       vmulsd    %xmm1, %xmm0, %xmm0
  2.68 │       vmovq     %xmm0, %rax
  2.12 │       movabsq   $0x7fff800000000000, %rdx # imm = 0x7FFF800000000000
       │       andq      %rax, %rdx
  0.90 │       movabsq   $0x405f000000000001, %rax # imm = 0x405F000000000001
  0.50 │       cmpq      %rax, %rdx
```

The vector pow operation in MLIR is lowered to a vector LLVM intrinsic:

```
%716 = call <32 x float> @llvm.pow.v32f32 ...
```

which perhaps can't be legalized to a vector sequence of instructions?

There are a couple of things we can do here:
1. Lower vector.pow in MLIR by using https://github.com/llvm/llvm-project/blob/657dbb4c394daec626bccfcaddc5341b8b9fc14e/mlir/include/mlir/Dialect/Math/Transforms/Passes.h#LL23C1-L24C1
2. Investigate why this particular `llvm.pow.v32f32` is not lowered to vector operations by the LLVM backend.

Option 1 can be a good initial step but it's risky in terms of performance and numerical precision. There are plenty of target-specific patterns in the LLVM backend, including x86 and ARM, to lower pow depending of the ISA and the fast-math configuration. It's very unlikely that we can generate an efficient sequence of instructions for all our CPU targets using only option 1, let alone different fast-math configurations/precision requirements. I would recommend that we quickly look at option 2 and, if justified, file an issue against LLVM and then decide if option 1 is needed. My guess is that `v32f32` might not be supported by the pow lowering or might not be properly legalized (for AVX512, `v32f32` should be split into two `v16f32` instructions).

### PR DESCRIPTION
**Pull request by dcaballe**
Following the same approach as for other math ops. This will prevent the powf operation to be scalarized on CPU.

Fixes #13547

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ class PolynomialApproximationPass
     RewritePatternSet mathPatterns(&getContext());
     populateExpandTanPattern(mathPatterns);
     populateExpandExp2FPattern(mathPatterns);
+    populateExpandPowFPattern(mathPatterns);
```
Comment (dcaballe) [inline compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp:33]: It's not clear to me what qualifies the expansions to be here or in the if-else statement below... but this should be similar to Exp2

**Comment by MaheshRavishankar** [comment]
Looks good, but waiting for benchmark runs

File: compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp (hunk 1 above)
Comment (bviyer) [reply compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp:33]: IIUC, it should be inside if-then-else if you are worried the approximation is not precise enough. As far as I know, if exp2 is precise enough then this shouldn't cause a huge issue.

**Review by bviyer — APPROVED** [review]
LGTM.

**Pull request merged**